R/title.R

```diff
@@ -0,0 +1,95 @@
+#' Generate Application Name
+#'
+#' Generate a short name (identifier) for an application given an application
+#' title.
+#'
+#' @param appTitle A descriptive title for the application.
+#' @param appPath The path to the application's content, either a directory
+#'   or an individual document. Optional.
+#' @param account The account where the application will be deployed. Optional.
+#' @return
+#' Returns a valid short name for the application.
+#'
+#' @details
+#' This function modifies the title until it forms a suitable application name.
+#' Suitable application names are 3 - 64 characters long and contain only
+#' alphanumeric characters.
+#'
+#' The function is intended to be used to find a name for a new application.
+#' If \code{appPath} and \code{account} are both specified, then the returned
+#' name will also be unique among locally known deployments of the directory
+#' (note that it is not guaranteed to be unique on the server).
+#'
+#' @examples
+#' # Generate a short name for a sample application
+#' generateAppName("My Father's Country", "~/fathers-country", "myacct")
+#'
+#' @export
+
+generateAppName <- function(appTitle, appPath = NULL, account = NULL) {
```

Comment:
Yes, most of it is. I've added some tests: https://github.com/rstudio/rsconnect/commit/d93bcacada88b8ec53425288a4ff9ec0fcf50c41